clippy_lints/src/manual_unwrap_or.rs

```diff
@@ -44,11 +44,34 @@ declare_clippy_lint! {
 declare_lint_pass!(ManualUnwrapOr => [MANUAL_UNWRAP_OR]);
 
 impl LateLintPass<'_> for ManualUnwrapOr {
-    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>) {
-        if in_external_macro(cx.sess(), expr.span) {
+    fn check_fn(
+        &mut self,
+        cx: &LateContext<'tcx>,
+        kind: FnKind<'tcx>,
+        _: &'tcx FnDecl<'tcx>,
+        body: &'tcx Body<'tcx>,
+        span: Span,
+        _: HirId,
+    ) {
+        if in_external_macro(cx.sess(), span) {
             return;
         }
-        lint_manual_unwrap_or(cx, expr);
+        if_chain! {
+            if let FnKind::ItemFn(_, _, header, _) = kind;
+            if !header.is_const();
```

Comment:
I think it would be better to use `in_constant` in this case: <https://github.com/rust-lang/rust-clippy/blob/1a206fc4abae0b57a3f393481367cf3efca23586/clippy_utils/src/lib.rs#L199>